File: code/scenes/src/main/java/org/fbme/scenes/controllers/DefaultFocusModel.java
package org.fbme.scenes.controllers;

import java.util.HashSet;

public class DefaultFocusModel implements SceneFocusModel {

    private final HashSet<FocusHandle> myHandles = new HashSet<>();

    @Override
    public void addFocus(FocusHandle focus) {
        myHandles.add(focus);
    }

    @Override
    public void clearFocus() {
        for (FocusHandle handle : myHandles) {
            handle.onFocusLost();
        }
        myHandles.clear();
    }
}

File: code/scenes/src/main/kotlin/org/fbme/scenes/controllers/DefaultFocusModel.kt
package org.fbme.scenes.controllers

class DefaultFocusModel : SceneFocusModel {
    private val handles = HashSet<FocusHandle>()

    override fun addFocus(focus: FocusHandle) {
        handles.add(focus)
    }

    override fun clearFocus() {
        for (handle in handles) {
            handle.onFocusLost()
        }
        handles.clear()
    }
}
